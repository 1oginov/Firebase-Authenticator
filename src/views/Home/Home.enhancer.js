// @flow

import { connect } from 'react-redux';
import { compose, withHandlers, type HOC } from 'recompose';

import { selectFirebaseApp as selectFirebaseAppCreator } from '../../actions';
import withNavigationHandlers from '../../enhancers/withNavigationHandlers';
import * as R from '../../routes';

const mapStateToProps = ({ firebaseApps }) => ({
  apps: Object.keys(firebaseApps).map(key => firebaseApps[key]),
});

const mapDispatchToProps = { selectFirebaseApp: selectFirebaseAppCreator };

const enhancer: HOC<*, {}> = compose(
  withNavigationHandlers({
    handleCreateFirebaseAppClick: R.CREATE_FIREBASE_APP,
    handleFirebaseAppNavigate: R.FIREBASE_APP,
  }),
  connect(mapStateToProps, mapDispatchToProps),
  withHandlers({

    onSelect: ({ handleFirebaseAppNavigate, selectFirebaseApp }) => (id) => {
      selectFirebaseApp(id);
      handleFirebaseAppNavigate();
    },

  }),
);

export default enhancer;
