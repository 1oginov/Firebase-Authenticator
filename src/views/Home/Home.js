// @flow

import AddIcon from '@material-ui/icons/Add';
import Fab from '@material-ui/core/Fab';
import * as React from 'react';

import FirebaseAppsList from '../../components/FirebaseAppsList';
import type { FirebaseApp } from '../../lib/firebaseApp';

type Props = {
  apps: Array<FirebaseApp>,
  classes: { [string]: string },
  handleCreateFirebaseAppClick: () => void,
  onSelect: string => void,
};

const Home = ({
  apps, classes, handleCreateFirebaseAppClick, onSelect,
}: Props) => (
  <div className={classes.root}>

    <FirebaseAppsList apps={apps} className={classes.firebaseAppsList} onSelect={onSelect} />

    <div className={classes.fabContainer}>
      <Fab onClick={handleCreateFirebaseAppClick} variant="extended">
        <AddIcon className={classes.fabIcon} />
        Create Firebase app
      </Fab>
    </div>

  </div>
);

export default Home;
