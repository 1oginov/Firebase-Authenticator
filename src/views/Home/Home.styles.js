// @flow

import { withStyles, type Theme } from '@material-ui/core/styles';

import wrapView from '../../style/wrapView';

const styles = (theme: Theme) => ({
  fabContainer: {
    textAlign: 'center',
  },
  fabIcon: {
    marginRight: theme.spacing.unit,
  },
  firebaseAppsList: {
    marginBottom: 3 * theme.spacing.unit,
  },
  root: wrapView(theme),
});

export default withStyles(styles);
