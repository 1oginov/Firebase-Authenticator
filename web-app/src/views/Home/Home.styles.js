import { withStyles, type Theme } from '@material-ui/core/styles';

import wrapView from '../../style/wrapView';

const styles = (theme: Theme) => ({
  fabContainer: {
    marginBottom: theme.spacing(3),
    textAlign: 'center',
  },
  fabIcon: {
    marginRight: theme.spacing(),
  },
  firebaseAppsList: {
    marginBottom: theme.spacing(3),
  },
  logo: {
    marginBottom: theme.spacing(5),
    marginTop: theme.spacing(2),
  },
  root: wrapView(theme),
});

export default withStyles(styles);
