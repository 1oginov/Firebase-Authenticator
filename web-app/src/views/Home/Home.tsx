import AddIcon from '@material-ui/icons/Add';
import Fab from '@material-ui/core/Fab';
import * as React from 'react';

import Disclaimer from '../../components/Disclaimer';
import FirebaseAppsList from '../../components/FirebaseAppsList';
import Logo from '../../components/Logo';
import type { FirebaseApp } from '../../lib/firebaseApp';

type Props = {
  apps: Array<FirebaseApp>,
  classes: { [string]: string },
  handleCreateFirebaseAppClick: () => void,
  onSelect: string => void,
};

const Home = ({
  apps, classes, handleCreateFirebaseAppClick, onSelect,
}: Props) => (
  <div className={classes.root}>

    <Logo className={classes.logo} />

    <FirebaseAppsList apps={apps} className={classes.firebaseAppsList} onSelect={onSelect} />

    <div className={classes.fabContainer}>
      <Fab color="secondary" onClick={handleCreateFirebaseAppClick} variant="extended">
        <AddIcon className={classes.fabIcon} />
        Create Firebase app
      </Fab>
    </div>

    <Disclaimer />

  </div>
);

export default Home;
