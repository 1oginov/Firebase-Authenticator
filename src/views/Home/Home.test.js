// @flow

import classes from 'identity-obj-proxy';
import { shallow } from 'enzyme';
import React from 'react';

import Home from './Home';
import app from '../../lib/__test-data__/firebaseApp';

describe('<Home />', () => {
  it('renders without crashing and matches snapshot', () => {
    const wrapper = shallow(
      <Home
        apps={[app]}
        classes={classes}
        handleCreateFirebaseAppClick={() => undefined}
        onSelect={() => undefined}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
