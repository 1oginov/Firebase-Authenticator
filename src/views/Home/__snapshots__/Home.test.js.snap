// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders without crashing and matches snapshot 1`] = `
<div
  className="root"
>
  <WithStyles(Logo)
    className="logo"
  />
  <WithStyles(FirebaseAppsList)
    apps={
      Array [
        Object {
          "config": Object {
            "apiKey": "API key",
            "authDomain": "Auth domain",
            "databaseURL": "Database URL",
            "messagingSenderId": "Messaging sender ID",
            "projectId": "Project ID",
            "storageBucket": "Storage bucket",
          },
          "id": "123",
          "redirect": Object {
            "refreshTokenPlaceholder": "%refreshToken%",
            "url": "https://example.com/auth?refresh-token=%refreshToken%",
          },
          "signInOptions": Array [
            "email",
            "github",
            "google",
          ],
          "title": "App",
        },
      ]
    }
    className="firebaseAppsList"
    onSelect={[Function]}
  />
  <div
    className="fabContainer"
  >
    <WithStyles(ForwardRef(Fab))
      color="secondary"
      onClick={[Function]}
      variant="extended"
    >
      <AddIcon
        className="fabIcon"
      />
      Create Firebase app
    </WithStyles(ForwardRef(Fab))>
  </div>
  <WithStyles(Disclaimer) />
</div>
`;
